### ISSUE TITLE
AttributeError: module 'pybigtools' has no attribute 'BBIReadError'

### ISSUE THREAD

**Issue by mtvector** [aertslab/CREsted#67]
### Report

Hi there,
Thanks for putting together this awesome looking package. I'm having this issue with pybigtools on initial install after following the installation instructions. 

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File [~/Matthew/utils/miniforge3/envs/crested/lib/python3.12/site-packages/crested/_io.py:455](https://aidc-ondemand-prd.corp.alleninstitute.org/node/n69/36772/lab/tree/Matthew/code/hmba-crossspecies-v1/analysis/atac/Matthew/utils/miniforge3/envs/crested/lib/python3.12/site-packages/crested/_io.py#line=454), in import_bigwigs(bigwigs_folder, regions_file, chromsizes_file, target, target_region_width, compress)
    453 try:
    454     # Validate using pyBigTools (add specific validation if available)
--> 455     bw = pybigtools.open(file_path, "r")
    456     bw_files.append(file_path)

AttributeError: module 'pybigtools' has no attribute 'open'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
Cell In[4], line 1
----> 1 crested.import_bigwigs(bigwigs_folder=atac_bigwig_dir,regions_file='[/allen/programs/celltypes/workgroups/rnaseqanalysis/EvoGen/Team/Matthew/data/cortex_data/internal_mouse/atac/merged_peaks.bed](https://aidc-ondemand-prd.corp.alleninstitute.org/allen/programs/celltypes/workgroups/rnaseqanalysis/EvoGen/Team/Matthew/data/cortex_data/internal_mouse/atac/merged_peaks.bed)',chromsizes_file=genome.sizes_file)

File [~/Matthew/utils/miniforge3/envs/crested/lib/python3.12/site-packages/crested/_io.py:460](https://aidc-ondemand-prd.corp.alleninstitute.org/node/n69/36772/lab/tree/Matthew/code/hmba-crossspecies-v1/analysis/atac/Matthew/utils/miniforge3/envs/crested/lib/python3.12/site-packages/crested/_io.py#line=459), in import_bigwigs(bigwigs_folder, regions_file, chromsizes_file, target, target_region_width, compress)
    458     except ValueError:
    459         pass
--> 460     except pybigtools.BBIReadError:
    461         pass
    463 bw_files = sorted(bw_files)

AttributeError: module 'pybigtools' has no attribute 'BBIReadError'
```

### Version information

```
-----
crested             1.1.0
genomepy            0.16.1
h5py                3.12.1
pandas              2.2.3
session_info        1.0.0
torch               2.5.1
-----
PIL                 11.0.0
absl                NA
anndata             0.11.1
appdirs             1.4.4
asttokens           NA
biothings_client    0.3.1
brotli              1.1.0
certifi             2024.08.30
cffi                1.17.1
charset_normalizer  3.4.0
colorama            0.4.6
comm                0.2.2
cycler              0.12.1
cython_runtime      NA
dateutil            2.9.0.post0
debugpy             1.8.8
decorator           5.1.1
diskcache           5.6.3
distutils           3.12.7
executing           2.1.0
filelock            3.16.1
fontTools           4.55.0
fsspec              2024.10.0
gmpy2               2.1.5
hdf5plugin          5.0.0
idna                3.10
igraph              0.11.8
ipykernel           6.29.5
jaraco              NA
jedi                0.19.2
joblib              1.4.2
keras               3.6.0
kiwisolver          1.4.7
leidenalg           0.10.2
llvmlite            0.43.0
logomaker           NA
loguru              0.7.2
matplotlib          3.9.2
ml_dtypes           0.5.0
modiscolite         2.2.1
more_itertools      10.3.0
mpl_toolkits        NA
mpmath              1.3.0
mygene              3.2.2
mysql               NA
namex               NA
natsort             8.4.0
norns               NA
numba               0.60.0
numpy               2.0.1
optree              0.13.1
packaging           24.2
parso               0.8.4
pickleshare         0.7.5
pkg_resources       NA
platformdirs        4.3.6
pooch               v1.8.2
prompt_toolkit      3.0.48
psutil              6.1.0
pure_eval           0.2.3
pyBigWig            0.3.23
pybigtools          NA
pycparser           2.22
pydev_ipython       NA
pydevconsole        NA
pydevd              3.2.2
pydevd_file_utils   NA
pydevd_plugins      NA
pydevd_tracing      NA
pyfaidx             0.8.1.3
pygments            2.18.0
pyparsing           3.2.0
pysam               0.22.1
pytz                2024.2
requests            2.32.3
rich                NA
scipy               1.14.1
seaborn             0.13.2
setuptools          75.5.0
six                 1.16.0
sklearn             1.5.2
socks               1.7.1
stack_data          0.6.2
sympy               1.13.1
texttable           1.7.0
threadpoolctl       3.5.0
torchgen            NA
tornado             6.4.1
tqdm                4.67.0
traitlets           5.14.3
triton              3.1.0
typing_extensions   NA
urllib3             2.2.3
wcwidth             0.2.13
yaml                6.0.2
zmq                 26.2.0
zstandard           0.23.0
-----
IPython             8.29.0
jupyter_client      8.6.3
jupyter_core        5.7.2
-----
Python 3.12.7 | packaged by conda-forge | (main, Oct  4 2024, 16:05:46) [GCC 13.3.0]
Linux-4.18.0-477.27.1.el8_8.x86_64-x86_64-with-glibc2.28
-----
Session information updated at 2024-11-22 13:36
```

conda list says:

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                  2_kmp_llvm    conda-forge
absl-py                   2.1.0                    pypi_0    pypi
anndata                   0.11.1                   pypi_0    pypi
aom                       3.6.1                h59595ed_0    conda-forge
appdirs                   1.4.4                    pypi_0    pypi
array-api-compat          1.9.1                    pypi_0    pypi
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
biopython                 1.84                     pypi_0    pypi
biothings-client          0.3.1                    pypi_0    pypi
blas                      2.116                       mkl    conda-forge
blas-devel                3.9.0            16_linux64_mkl    conda-forge
brotli-python             1.1.0           py312h2ec8cdc_2    conda-forge
bzip2                     1.0.8                h4bc722e_7    conda-forge
c-ares                    1.32.3               h4bc722e_0    conda-forge
ca-certificates           2024.8.30            hbcca054_0    conda-forge
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
cffi                      1.17.1          py312h06ac9bb_0    conda-forge
charset-normalizer        3.4.0              pyhd8ed1ab_0    conda-forge
click                     8.1.7                    pypi_0    pypi
cmake                     3.31.0.1                 pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
comm                      0.2.2              pyhd8ed1ab_0    conda-forge
contourpy                 1.3.1                    pypi_0    pypi
cpython                   3.12.7          py312hd8ed1ab_0    conda-forge
crested                   1.1.0                    pypi_0    pypi
cuda-cudart               12.4.127                      0    nvidia
cuda-cupti                12.4.127                      0    nvidia
cuda-libraries            12.4.1                        0    nvidia
cuda-nvrtc                12.4.127                      0    nvidia
cuda-nvtx                 12.4.127                      0    nvidia
cuda-opencl               12.6.77                       0    nvidia
cuda-runtime              12.4.1                        0    nvidia
cuda-version              12.6                          3    nvidia
cycler                    0.12.1                   pypi_0    pypi
debugpy                   1.8.8           py312h2ec8cdc_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
diskcache                 5.6.3                    pypi_0    pypi
docker-pycreds            0.4.0                    pypi_0    pypi
exceptiongroup            1.2.2              pyhd8ed1ab_0    conda-forge
executing                 2.1.0              pyhd8ed1ab_0    conda-forge
ffmpeg                    4.4.2           gpl_hdf48244_113    conda-forge
filelock                  3.16.1             pyhd8ed1ab_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_3    conda-forge
fontconfig                2.15.0               h7e30c49_1    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.55.0                   pypi_0    pypi
freetype                  2.12.1               h267a509_2    conda-forge
fsspec                    2024.10.0                pypi_0    pypi
genomepy                  0.16.1                   pypi_0    pypi
gettext                   0.22.5               he02047a_3    conda-forge
gettext-tools             0.22.5               he02047a_3    conda-forge
giflib                    5.2.2                hd590300_0    conda-forge
gitdb                     4.0.11                   pypi_0    pypi
gitpython                 3.1.43                   pypi_0    pypi
gmp                       6.3.0                hac33072_2    conda-forge
gmpy2                     2.1.5           py312h7201bc8_2    conda-forge
gnutls                    3.7.9                hb077bed_0    conda-forge
h2                        4.1.0              pyhd8ed1ab_0    conda-forge
h5py                      3.12.1                   pypi_0    pypi
hdf5plugin                5.0.0                    pypi_0    pypi
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
idna                      3.10               pyhd8ed1ab_0    conda-forge
igraph                    0.11.8                   pypi_0    pypi
importlib-metadata        8.5.0              pyha770c72_0    conda-forge
ipykernel                 6.29.5             pyh3099207_0    conda-forge
ipython                   8.29.0             pyh707e725_0    conda-forge
jedi                      0.19.2             pyhff2d567_0    conda-forge
jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge
joblib                    1.4.2                    pypi_0    pypi
jupyter_client            8.6.3              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2              pyh31011fe_1    conda-forge
keras                     3.6.0                    pypi_0    pypi
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.7                    pypi_0    pypi
krb5                      1.21.3               h659f571_0    conda-forge
lame                      3.100             h166bdaf_1003    conda-forge
lcms2                     2.16                 hb7c19ff_0    conda-forge
ld_impl_linux-64          2.43                 h712a8e2_2    conda-forge
leidenalg                 0.10.2                   pypi_0    pypi
lerc                      4.0.0                h27087fc_0    conda-forge
libasprintf               0.22.5               he8f35ee_3    conda-forge
libasprintf-devel         0.22.5               he8f35ee_3    conda-forge
libblas                   3.9.0            16_linux64_mkl    conda-forge
libcblas                  3.9.0            16_linux64_mkl    conda-forge
libcublas                 12.4.5.8                      0    nvidia
libcufft                  11.2.1.3                      0    nvidia
libcufile                 1.11.1.6                      0    nvidia
libcurand                 10.3.7.77                     0    nvidia
libcurl                   8.10.1               hbbe4b11_0    conda-forge
libcusolver               11.6.1.9                      0    nvidia
libcusparse               12.3.1.170                    0    nvidia
libdeflate                1.21                 h4bc722e_0    conda-forge
libdrm                    2.4.123              hb9d3cd8_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libegl                    1.7.0                ha4b6fd6_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libexpat                  2.6.4                h5888daf_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc                    14.2.0               h77fa898_1    conda-forge
libgcc-ng                 14.2.0               h69a702a_1    conda-forge
libgettextpo              0.22.5               he02047a_3    conda-forge
libgettextpo-devel        0.22.5               he02047a_3    conda-forge
libgfortran               14.2.0               h69a702a_1    conda-forge
libgfortran-ng            14.2.0               h69a702a_1    conda-forge
libgfortran5              14.2.0               hd5240d6_1    conda-forge
libgl                     1.7.0                ha4b6fd6_2    conda-forge
libglvnd                  1.7.0                ha4b6fd6_2    conda-forge
libglx                    1.7.0                ha4b6fd6_2    conda-forge
libgomp                   14.2.0               h77fa898_1    conda-forge
libhwloc                  2.11.1          default_hecaa2ac_1000    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libidn2                   2.3.7                hd590300_0    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0            16_linux64_mkl    conda-forge
liblapacke                3.9.0            16_linux64_mkl    conda-forge
libnghttp2                1.64.0               h161d5f1_0    conda-forge
libnpp                    12.2.5.30                     0    nvidia
libnsl                    2.0.1                hd590300_0    conda-forge
libnvfatbin               12.6.77                       0    nvidia
libnvjitlink              12.4.127                      0    nvidia
libnvjpeg                 12.3.1.117                    0    nvidia
libpciaccess              0.18                 hd590300_0    conda-forge
libpng                    1.6.44               hadc24fc_0    conda-forge
libsodium                 1.0.20               h4ab18f5_0    conda-forge
libsqlite                 3.47.0               hadc24fc_1    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge
libstdcxx-ng              14.2.0               h4852527_1    conda-forge
libtasn1                  4.19.0               h166bdaf_0    conda-forge
libtiff                   4.7.0                h6565414_0    conda-forge
libunistring              0.9.10               h7f98852_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libva                     2.22.0               h8a09558_1    conda-forge
libvpx                    1.13.1               h59595ed_0    conda-forge
libwebp                   1.4.0                h2c329e2_0    conda-forge
libwebp-base              1.4.0                hd590300_0    conda-forge
libxcb                    1.17.0               h8a09558_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxml2                   2.13.5               h064dc61_0    conda-forge
libzlib                   1.3.1                hb9d3cd8_2    conda-forge
llvm-openmp               15.0.7               h0cdce71_0    conda-forge
llvmlite                  0.43.0                   pypi_0    pypi
logomaker                 0.8                      pypi_0    pypi
loguru                    0.7.2                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                3.0.2           py312h178313f_0    conda-forge
matplotlib                3.9.2                    pypi_0    pypi
matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge
mdurl                     0.1.2                    pypi_0    pypi
mkl                       2022.1.0           h84fe81f_915    conda-forge
mkl-devel                 2022.1.0           ha770c72_916    conda-forge
mkl-include               2022.1.0           h84fe81f_915    conda-forge
ml-dtypes                 0.5.0                    pypi_0    pypi
modisco-lite              2.2.1                    pypi_0    pypi
mpc                       1.3.1                h24ddda3_1    conda-forge
mpfr                      4.2.1                h90cbb55_3    conda-forge
mpmath                    1.3.0              pyhd8ed1ab_0    conda-forge
mygene                    3.2.2                    pypi_0    pypi
mysql-connector-python    9.1.0                    pypi_0    pypi
namex                     0.0.8                    pypi_0    pypi
natsort                   8.4.0                    pypi_0    pypi
ncurses                   6.5                  he02047a_1    conda-forge
nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge
nettle                    3.9.1                h7ab15ed_0    conda-forge
networkx                  3.4.2              pyh267e887_2    conda-forge
norns                     0.1.6                    pypi_0    pypi
nose                      1.3.7                    pypi_0    pypi
numba                     0.60.0                   pypi_0    pypi
numpy                     2.0.1                    pypi_0    pypi
openh264                  2.3.1                hcb278e6_2    conda-forge
openjpeg                  2.5.2                h488ebb8_0    conda-forge
openssl                   3.4.0                hb9d3cd8_0    conda-forge
optree                    0.13.1                   pypi_0    pypi
p11-kit                   0.24.1               hc5aa10d_0    conda-forge
packaging                 24.2               pyhff2d567_1    conda-forge
pandas                    2.2.3                    pypi_0    pypi
parso                     0.8.4              pyhd8ed1ab_0    conda-forge
pexpect                   4.9.0              pyhd8ed1ab_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    11.0.0          py312h7b63e92_0    conda-forge
pip                       24.3.1             pyh8b19718_0    conda-forge
platformdirs              4.3.6              pyhd8ed1ab_0    conda-forge
pooch                     1.8.2                    pypi_0    pypi
prompt-toolkit            3.0.48             pyha770c72_0    conda-forge
protobuf                  5.28.3                   pypi_0    pypi
psutil                    6.1.0           py312h66e93f0_0    conda-forge
pthread-stubs             0.4               hb9d3cd8_1002    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.3              pyhd8ed1ab_0    conda-forge
pybigtools                0.2.2                    pypi_0    pypi
pybigwig                  0.3.23                   pypi_0    pypi
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pyfaidx                   0.8.1.3                  pypi_0    pypi
pygments                  2.18.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.2.0                    pypi_0    pypi
pysam                     0.22.1          py312hcfdcdd7_2    bioconda
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.12.7          hc5c86c4_0_cpython    conda-forge
python-dateutil           2.9.0.post0        pyhff2d567_0    conda-forge
python_abi                3.12                    5_cp312    conda-forge
pytorch                   2.5.1           py3.12_cuda12.4_cudnn9.1.0_0    pytorch
pytorch-cuda              12.4                 hc786d27_7    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2024.2                   pypi_0    pypi
pyyaml                    6.0.2           py312h66e93f0_1    conda-forge
pyzmq                     26.2.0          py312hbf22597_3    conda-forge
readline                  8.2                  h8228510_1    conda-forge
requests                  2.32.3             pyhd8ed1ab_0    conda-forge
rich                      13.9.4                   pypi_0    pypi
scikit-learn              1.5.2                    pypi_0    pypi
scipy                     1.14.1                   pypi_0    pypi
seaborn                   0.13.2                   pypi_0    pypi
sentry-sdk                2.18.0                   pypi_0    pypi
session-info              1.0.0                    pypi_0    pypi
setproctitle              1.3.4                    pypi_0    pypi
setuptools                75.5.0             pyhff2d567_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
smmap                     5.0.1                    pypi_0    pypi
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
stdlib-list               0.11.0                   pypi_0    pypi
svt-av1                   1.4.1                hcb278e6_0    conda-forge
sympy                     1.13.1                   pypi_0    pypi
tangermeme                0.4.2                    pypi_0    pypi
tbb                       2021.13.0            h84d6215_0    conda-forge
texttable                 1.7.0                    pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
tk                        8.6.13          noxft_h4845f30_101    conda-forge
torchaudio                2.5.1               py312_cu124    pytorch
torchtriton               3.1.0                     py312    pytorch
torchvision               0.20.1              py312_cu124    pytorch
tornado                   6.4.1           py312h66e93f0_1    conda-forge
tqdm                      4.67.0                   pypi_0    pypi
traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge
typing_extensions         4.12.2             pyha770c72_0    conda-forge
tzdata                    2024.2                   pypi_0    pypi
urllib3                   2.2.3              pyhd8ed1ab_0    conda-forge
wandb                     0.18.7                   pypi_0    pypi
wayland                   1.23.1               h3e06ad9_0    conda-forge
wayland-protocols         1.37                 hd8ed1ab_0    conda-forge
wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge
wheel                     0.45.0             pyhd8ed1ab_0    conda-forge
x264                      1!164.3095           h166bdaf_2    conda-forge
x265                      3.5                  h924138e_3    conda-forge
xorg-libx11               1.8.10               h4f16b4b_0    conda-forge
xorg-libxau               1.0.11               hb9d3cd8_1    conda-forge
xorg-libxdmcp             1.1.5                hb9d3cd8_0    conda-forge
xorg-libxext              1.3.6                hb9d3cd8_0    conda-forge
xorg-libxfixes            6.0.1                hb9d3cd8_0    conda-forge
xorg-xorgproto            2024.1               hb9d3cd8_1    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
zeromq                    4.3.5                h3b0a872_7    conda-forge
zipp                      3.21.0             pyhd8ed1ab_0    conda-forge
zstandard                 0.23.0          py312hef9b889_1    conda-forge
zstd                      1.5.6                ha6fb4c9_0    conda-forge
```

**Comment by UCDNJJ** [comment]
@mtvector I fixed this by installing from the conda repo: `conda install -c bioconda pybigtools`. Not sure what's wrong with the pip version.

**Comment by LukasMahieu** [comment]
Can't reproduce this myself unfortunately. I'll raise an issue on the pybigtools github and see if they know what's up.
Thanks for letting us know.

**Comment by mtvector** [comment]
> @mtvector I fixed this by installing from the conda repo: `conda install -c bioconda pybigtools`. Not sure what's wrong with the pip version.

Thanks for raising the issue with the pybigtools folks, I can report that installing the conda pybigtools fixed the error for me as well!

**Comment by jackh726** [comment]
This issue should be fixed started with pybigtools 0.2.3 that I released today. There was an issue with the sdist to pypi.

**Comment by LukasMahieu** [comment]
Thanks for the fix!
